src/main/java/chopchop/logic/commands/StatsIngredientDateCommand.java

```diff
@@ -0,0 +1,61 @@
+package chopchop.logic.commands;
+
+import static chopchop.commons.util.Strings.ARG_AFTER;
+import static chopchop.commons.util.Strings.ARG_BEFORE;
+import static chopchop.commons.util.Strings.ARG_ON;
+
+import java.time.LocalDateTime;
+
+import chopchop.logic.commands.exceptions.CommandException;
+import chopchop.logic.history.HistoryManager;
+import chopchop.model.Model;
+
+public class StatsIngredientDateCommand extends Command {
+    public static final String COMMAND_WORD = "stats ingredient";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Shows a list of ingredient. "
+        + "Parameters: "
+        + "[" + ARG_ON + " DATE] "
+        + "[" + ARG_BEFORE + "DATE] "
+        + "[" + ARG_AFTER + "DATE] "
+        + "Example " + COMMAND_WORD + " " + ARG_BEFORE + " 2020-02-13";
+
+    private final LocalDateTime on;
+    private final LocalDateTime before;
+    private final LocalDateTime after;
+
+    /**
+     * Creates an AddIngredientCommand to add the specified {@code Ingredient}
+     */
+    public StatsIngredientDateCommand(LocalDateTime on, LocalDateTime before,
+                                LocalDateTime after) {
+        assert (on == null && (before != null || after != null))
+            || (on != null && (before == null && after == null));
```

Comment:
feels a bit dubious but maybe you should make static methods instead to handle these cases?

Reply:
I think I will just make 'on' take precedence over 'before' and 'after'. So if allNonNull then only on is considered.
I will think about other better ways.